### PR TITLE
Added methods for async clients

### PR DESCRIPTION
**Pull request by gabfssilva**
I added a few methods for async clients java api and a test for SNS too.

Not sure if I missed something, but, waiting for a feedback so I can improve this PR.

### CODE REVIEW

File: localstack/ext/java/src/main/java/cloud/localstack/TestUtils.java [hunk 1]
```diff
@@ -43,12 +47,57 @@ public static AmazonSQS getClientSQS() {
                 withCredentials(getCredentialsProvider()).build();
     }
 
+    public static AmazonSQSAsync getClientSQSAsync() {
```
Comment (whummer) [inline localstack/ext/java/src/main/java/cloud/localstack/TestUtils.java]: nitpick: can we change this method to:
```
    public static AmazonSQSAsync getClientSQSAsync() {
        return getClientSQSAsync(null);
    }
```
This way we can avoid code duplication and only maintain the logic in a single method. (same for `getClientSNSAsync`, `getClientLambdaAsync`, and `getClientKinesisAsync`)

**Comment by whummer** [comment]
Thanks for this PR @gabfssilva . Any chance we can address the nitpick comment before merging?

**Comment by whummer** [comment]
Trying one more time - any updates on this PR @gabfssilva ? thanks

**Comment by gabfssilva** [comment]
Sorry for the delay. I'll commit the changes soon. 

[]'s

**Comment by whummer** [comment]
Any updates @gabfssilva ? Please let us know if you're still planning to incorporate these changes. Please also rebase your branch onto latest `master` to fix the merge conflict. Thanks a lot

**Comment by ax-slad** [comment]
Any chance we can get this merged soon? Most of AWS integration libraries now use async client so it is not possible to use Junit runner with local stack config now.

**Comment by netstart** [comment]
> Any chance we can get this merged soon? Most of AWS integration libraries now use async client so it is not possible to use Junit runner with local stack config now.

For now I use something like this:

https://gist.github.com/netstart/d666f617196c45c826cfae298664deaf

**Pull request merged**